src/pages/LogoOrganizations.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { ChangeEvent, Component } from "react";
+
+import { WithNamespaces, withNamespaces } from "react-i18next";
+
+import { StorageContext } from "../context/storage";
+import { putToBackend } from "../utils/backend";
+
+import * as ts from "io-ts";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { OrganizationFiscalCode } from "../../generated/definitions/api/OrganizationFiscalCode";
+import Logo from "../components/Logo";
+import { getConfig } from "../utils/config";
+import { getBase64OfImage } from "../utils/image";
+
+import { Input } from "design-react-kit";
+
+const LogoParamsApi = ts.interface({
+  logo: NonEmptyString,
+  organizationFiscalCode: OrganizationFiscalCode
+});
+
+type LogoParamsApi = ts.TypeOf<typeof LogoParamsApi>;
+
+const LogoErrorBodyApi = ts.interface({
+  detail: ts.string,
+  status: ts.number,
+  title: ts.string
+});
+
+type LogoErrorBodyApi = ts.TypeOf<typeof LogoErrorBodyApi>;
+
+const LogoSuccessBodyApi = ts.interface({});
+
+type LogoSuccessBodyApi = ts.TypeOf<typeof LogoSuccessBodyApi>;
+
+const LOGO_PATH = getConfig("IO_LOGO_PATH") + "/services/";
+
+type OwnProps = {};
+type Props = WithNamespaces & OwnProps;
+
+type LogoState = {
+  errorLogoUpload: boolean;
+  logo?: string;
+  logoIsValid: boolean;
+  logoUploaded: boolean;
+  organizationFiscalCode?: OrganizationFiscalCode;
+  organizationFiscalCodeIsValid: boolean;
+};
+
+class LogoOrganizations extends Component<Props, LogoState> {
+  public state: LogoState = {
+    errorLogoUpload: false,
+    logo: undefined,
+    logoIsValid: true,
+    logoUploaded: false,
+    organizationFiscalCode: undefined,
+    organizationFiscalCodeIsValid: true
+  };
+
+  public handleServiceLogoSubmit = async () => {
+    LogoParamsApi.decode({
+      logo: this.state.logo,
+      organizationFiscalCode: this.state.organizationFiscalCode
+    }).fold(
+      _ =>
+        this.setState({
+          logo: undefined,
+          logoIsValid: false,
+          logoUploaded: false
+        }),
+      async logoParamsApi => {
+        const responseOrError = await putToBackend({
+          path: `organizations/${logoParamsApi.organizationFiscalCode}/logo`,
+          options: {
+            body: JSON.stringify({ logo: logoParamsApi.logo })
+          }
+        });
+
+        LogoErrorBodyApi.decode(responseOrError).bimap(
+          () =>
+            LogoSuccessBodyApi.decode(responseOrError).map(_ => {
+              this.setState({
+                errorLogoUpload: false,
+                logo: undefined,
+                logoIsValid: true,
+                logoUploaded: true
+              });
+            }),
+          _ =>
+            this.setState({
+              errorLogoUpload: true,
+              logo: undefined,
+              logoIsValid: false,
+              logoUploaded: false
+            })
+        );
+      }
+    );
```

Comment:
Can we collapse this mapping operations? i.e:
```
await putToBackend({
            path: `organizations/${logoParamsApi.organizationFiscalCode}/logo`,
            options: {
              body: JSON.stringify({ logo: logoParamsApi.logo })
            }
          }).map(LogoErrorBodyApi.decode).bimap(...)
```

Reply:
I tried but I don't know how to solve, because there are two constraints:
- there is no _fp-ts_ library for this project (so no `Either`, `TaskEither` ...) but only decode/encode from _io-utils_;
- putToBackend is defined as follow : 
                  `putToBackend<T>(params: PutToBackendParams): Promise<T>`

Reply:
consider that this project is going to be completely refactored so it's not worth to spend a lot of time on it :-)

Reply:
In this case... LGTM @infantesimone :)